pypeit/core/coadd.py

```diff
@@ -549,29 +551,40 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     # Determine an initial guess
     ratio = robust_median_ratio(flux, ivar, flux_ref, ivar_ref, mask=mask, mask_ref=mask_ref,
                                 ref_percentile=ref_percentile, max_factor=scale_max)
+    wave_min = wave.min()
+    wave_max = wave.max()
+
     if 'poly' in model:
         guess = np.append(ratio, np.zeros(norder-1))
+        scale_fun = lambda x: x
     elif 'square' in model:
         guess = np.append(np.sqrt(ratio), np.zeros(norder-1))
+        scale_fun = np.sqrt
     elif 'exp' in model:
         guess = np.append(np.log(ratio), np.zeros(norder-1))
+        scale_fun = np.log
     else:
         msgs.error('Unrecognized model type')
 
-    wave_min = wave.min()
-    wave_max = wave.max()
 
     # Now compute median filtered versions of the spectra which we will actually operate on for the fitting. Note
     # that rejection will however work on the non-filtered spectra.
     med_width = (2.0*np.ceil(median_frac/2.0*nspec) + 1).astype(int)
     flux_med, ivar_med = median_filt_spec(flux, ivar, mask, med_width)
     flux_ref_med, ivar_ref_med = median_filt_spec(flux_ref, ivar_ref, mask_ref, med_width)
 
+    lr = scipy.stats.linregress(wave[mask], scale_fun(flux_ref_med[mask]))
+    if np.abs(lr.rvalue) > 0.8:
+        leg_slope = lr.slope * (wave_max - wave_min) / 2.0
+        leg_int = lr.intercept + lr.slope * (wave_max + wave_min)/2.0
+        guess = np.append([leg_int, leg_slope], np.zeros(norder-2))
```

Comment:
So why cut on the lr.rvalue? If all this is doing is a linear fit, shouldn't we be using the errors here. stats.linregress does not take errors, but there are other linear fitting codes that do.

Reply:
I'll look into cutting on something like reduced chi^2

Reply:
Okay, I've replaced stats.linregress with np.polyfit, which takes errors, and handle error propagation from the spectrum rescaling appropriately. I found that using a quadratic fit for the guess generally improved many of the cases where a linear guess was worse than a constant guess. I've also moved to cutting on reduced chi^2.